pythonPackages.aws-lambda-builders: init at 0.2.1

dhl opened the pull request:
A tool to compile, build and package AWS Lambda functions.

This is being packaged as part of the effort to update `aws-sam-cli`.

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

worldofpeace commented on pkgs/development/python-modules/aws-lambda-builders/default.nix:
```diff
@@ -0,0 +1,31 @@
+  ];
+
+  meta = with lib; {
+    homepage = https://github.com/awslabs/aws-serverlessrepo-python;
```
```suggestion
    homepage = https://github.com/awslabs/aws-lambda-builders;
```

worldofpeace commented on pkgs/development/python-modules/aws-lambda-builders/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  meta = with lib; {
+    homepage = https://github.com/awslabs/aws-serverlessrepo-python;
+    description = "A tool to compile, build and package AWS Lambda functions.";
```
```suggestion
    description = "A tool to compile, build and package AWS Lambda functions";
```

worldofpeace commented:
> This is being packaged as part of the effort to update aws-sam-cli.

It's usually preferred to open one pr for that kind of purpose. Even if you don't update `aws-sam-cli` in the pr.

worldofpeace commented on pkgs/development/python-modules/aws-lambda-builders/default.nix:
```diff
@@ -0,0 +1,31 @@
+  pname = "aws-lambda-builders";
+  version = "0.2.1";
+
+  src = fetchPypi {
```
I noticed that there's no tests distributed on `pypi`.

Can you use `fetchFromGitHub` to get an archive that has them and enable them?

dhl replied:
@worldofpeace I've enabled the tests but it looks like tests are trying to write files.

Is that allowed in the sandbox?

worldofpeace replied:
The build cannot write files outside of the build prefix.

Can you paste the errors in a gist?

dhl replied:
Here it is https://gist.github.com/dhl/bdc933473f1856015e550f14646936c0

dhl replied:
@worldofpeace managed to fix the tests by adding the package bin path in `preCheck` phase:

```
preCheck = "export PATH=$out/bin:$PATH";
```

worldofpeace reviewed (changes requested):
See the comments above.

dhl commented:
If the package looks good to you, I'll remove the change to `maintainers.nix` and squash the commits.

worldofpeace reviewed (changes requested):
Looking at their `Makefile` I think we should be running the tests like
```
pytest tests/functional
```

in the `checkPhase`.

worldofpeace commented:
It also looks like this shouldn't be built on `!=3.5.*` so we should use the `disabled` parameter to disable it for that interpreter version

```
disabled = isPy35
```

else build fails for `python35Packages.aws-lambda-builders`
```
aws-lambda-builders requires Python '>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*' but the running Python is 3.5.7
```

dhl committed:
pythonPackages.aws-lambda-builders: init at 0.2.1

dhl commented:
@worldofpeace I've added implemented all the suggested changes and squashed them:

1. Fix homepage URL
2. Remove trailing period in description
3. Fetch from GitHub to get tests
4. Run functional tests (unit and integration tests not working)
5. Disabled for python 3.5
6. Remove adding myself to `maintainers.nix` (merged in #59088)

worldofpeace reviewed (approved):
Builds for every interpreter version :sparkles:
